src/id_generator.py

```diff
@@ -0,0 +1,18 @@
+class IDGenerator:
+    def __init__(self):
+        self.current_id = 0
+        self.used_ids = []
+
+    def assign_id(self):
```

Comment:
For the name assign I would expect it to assign the Id to an object. Rather a new Id is just created and ready for use. Assignment w (to an object) will hopefully happen in calling code